### ISSUE TITLE
[KTX] onNeverAskAgain gets called when touch outside of permission dialog

### ISSUE THREAD

**Issue by TurKurT656** [permissions-dispatcher/PermissionsDispatcher#745]
## Overview

- When the permission dialog pops up, if user touches outside of the permission dialog, the `onNeverAskAgain` lambda gets executed instead on `onPermissionDenied`

### Expected

- Execution of `onPermissionDenied` when user touches outside the permission dialog

### Actual

- Execution of `onNeverAskAgain` when user touches outside the permission dialog

## Environment

- Which library version are you using? **1.1.2**
- On which devices do you observe the issue? **All**

## Reproducible steps

- Ask for a permission with `permission.launch()`
- After the dialog pops up, touch outside of the dialog's area

**Comment by hotchemi** [comment]
Sorry I'm not sure how to repro the issue, is my steps correct?
![1](https://user-images.githubusercontent.com/471318/132944468-0b8ed003-eb03-4386-94c5-0b59ea4b7efa.gif)

**Comment by TurKurT656** [comment]
can you test this on API 30?
This is a little bit strange cause i can easily dismiss the dialog by touching outside of the dialog

**Comment by hotchemi** [comment]
Thx, I could dismiss the dialogue but looks like onPermissionDenied is called. could you try it out with ktx-sample module?

**Comment by TurKurT656** [comment]
I tried **ktx-sample** and it was ok (never ask again doesn't called). However the first time I called **camera** permission and then I touched outside of the dialog neither `onNeverAskAgain` nor `onPermissionDenied` get called. But after I pressed deny button once and then launched the permission again and this time by pressing outside of the dialog `onPermissionDenied` gets called .

**Comment by hotchemi** [comment]
thx! um...somehow couldn't repro but let me try again.

**Comment by ParkJong-Hun** [comment]
[Memo] ↑ If `@OnNeverAskAgain` exists, a bug occurs, but if it does not exist, the `@OnPermissionDenied` function is called.
